models/yang/sonic/sonic-acl.yang

```diff
@@ -88,29 +82,38 @@ module sonic-acl {
 					type leafref {
 						path "../../../ACL_TABLE/ACL_TABLE_LIST/aclname";
 					}
-					must "(/cmn:operation/cmn:operation != 'DELETE') or " +
-						"count(current()/../../../ACL_TABLE/ACL_TABLE_LIST[aclname=current()]/ports) = 0" {
-							error-message "Ports are already bound to this rule.";
-						}
 				} 
 
 				leaf rulename {
-					type string;
+					type string {
+						pattern "[^ |:]+" {
+							error-app-tag rulename-invalid;
+						}
+						length 1..63 {
```

Comment:
The regex for rulename looks not correct. It could allow a special character like "?", "+", "&" etc.
You can also define the occurrence number in the regex. I would suggest the pattern like "[a-zA-Z0-9_-]{1,63}"